client/interface.go

```diff
@@ -0,0 +1,30 @@
+package client
+
+import "github.com/theupdateframework/go-tuf/data"
+
+type TUFClient interface {
+	// Update downloads and verifies remote metadata and returns updated targets.
+	// It always performs root update (5.2 and 5.3) section of the v1.0.19 spec.
+	Update() (data.TargetFiles, error)
+
+	// Download downloads the given target file from remote storage into dest.
+	//
+	// dest will be deleted and an error returned in the following situations:
+	//
+	//   - The target does not exist in the local targets.json
+	//   - Failed to fetch the chain of delegations accessible from local snapshot.json
+	//   - The target does not exist in any targets
+	//   - Metadata cannot be generated for the downloaded data
+	//   - Generated metadata does not match local metadata for the given file
+	Download(name string, dest Destination) (err error)
+
+	VerifyDigest(digest string, digestAlg string, length int64, path string) error
```

Comment:
why does this need to be exposed? Is this for verifying a target against some local metadata?

Reply:
> why does this need to be exposed? 

The unhelpful answer is that it's part of the effective public interface that a Client exposes and it's covered in `client_test.go` so if we want the whole test suite to pass we need to expose it 😛 

I think specifically it's for OCI and other, similar contexts: https://github.com/theupdateframework/go-tuf/pull/170

It feels like a useful part of the public interface generally IMO. It's already part of the `Client` struct so we're somewhat committed to it. I think if we have this method in `Client`, it should be part of the `TUFClient` interface. I

You can implement this by calling `Target(path)` and then checking the hash etc. manually; that seems slightly error-prone 🤷 I'd consider removing it from the struct and the interface, can't find anybody that uses it. But I think let's make that a separate discussion.

>  Is this for verifying a target against some local metadata?

Roughly, yes. @dibrinsofor could we add a comment like the following?

> VerifyDigest verifies a target at the given path by its digest.
>
> This is useful in contexts where the targets are distributed via some channel besides TUF (e.g., OCI registries).

Reply:
Got it, the linked PR was useful for context!

just for future reference testing in the same package doesn't require only using exported methods, you can test internal methods in the same package.

I do think it's useful, but I almost think my gripe was that the interface was kind of confusing. I'm supplying a hash, then alg, then size, and then path. Presumably of my target, and expect that this method will get the target the path and compare the metadata. 

I think VerifyTargetWithDigest(targetPath, digestAlg, digest, length) woulda made more sense, but we can work on it separately from this interface discusison